packages/cli/generation/ir-generator/src/IdGenerator.ts

```diff
@@ -24,9 +18,9 @@ export const IdGenerator = {
         const joinedFernFilePath = stringifyFernFilepath(fernFilepath);
         return `subpackage_${joinedFernFilePath}`;
     },
-    generateEndpointId: (declaredServiceName: DeclaredServiceName, httpEndpoint: HttpEndpoint): string => {
+    generateEndpointId: (declaredServiceName: DeclaredServiceName, httpEndpointName: Name): string => {
```

Comment:
can we pass the `HttpEndpoint` directly, a lot of things have `Name` so doesnt provide as much type safety

Reply:
I originally wanted to, but we actually use this function while we're constructing the `HttpEndpoint` itself, so it's actually easier to use the name here.

See `convertHttpService.ts` for reference:
```typescript
            const endpointName = file.casingsGenerator.generateName(endpointKey);
            return {
                ...convertDeclaration(endpoint),
                name: file.casingsGenerator.generateName(endpointKey),
                id: IdGenerator.generateEndpointId(serviceName, endpointName),
                name: endpointName,
                displayName: endpoint["display-name"],
                auth: endpoint.auth ?? serviceDefinition.auth,
                baseUrl: endpoint.url ?? serviceDefinition.url,
```

Reply:
got it you could make the type `Omit<HttpEndpoint, "id">` ? not the cleanest but i think better than name?